olp-cpp-sdk-dataservice-read/tests/QueryApiTest.cpp

```diff
@@ -0,0 +1,256 @@
+                                     kHttpResponseEmpty));
+    int32_t depth = 2;
+    auto index_response =
+        olp::dataservice::read::QueryApi::QuadTreeIndexVolatile(
+            *client_, {}, kQuadKey, depth, boost::none, boost::none,
+            olp::client::CancellationContext{});
+
+    ASSERT_FALSE(index_response.IsSuccessful());
+    auto error = index_response.GetError();
+    ASSERT_EQ(http::HttpStatusCode::BAD_REQUEST, error.GetHttpStatusCode());
+    ASSERT_EQ(kHttpResponseEmpty, error.GetMessage());
+  }
+  {
+    SCOPED_TRACE("Invalid quad key.");
+    EXPECT_CALL(*network_mock_, Send(_, _, _, _, _))
+        .WillOnce(ReturnHttpResponse(http::NetworkResponse().WithStatus(
+                                         http::HttpStatusCode::BAD_REQUEST),
+                                     kHttpResponseEmpty));
+    int32_t depth = 2;
+    auto index_response =
+        olp::dataservice::read::QueryApi::QuadTreeIndexVolatile(
+            *client_, kLayerid, {}, depth, boost::none, boost::none,
+            olp::client::CancellationContext{});
+
+    ASSERT_FALSE(index_response.IsSuccessful());
+    auto error = index_response.GetError();
+    ASSERT_EQ(http::HttpStatusCode::BAD_REQUEST, error.GetHttpStatusCode());
+    ASSERT_EQ(kHttpResponseEmpty, error.GetMessage());
+  }
+  {
+    SCOPED_TRACE("Invalid depth.");
+    EXPECT_CALL(*network_mock_, Send(_, _, _, _, _))
+        .WillOnce(ReturnHttpResponse(http::NetworkResponse().WithStatus(
+                                         http::HttpStatusCode::BAD_REQUEST),
+                                     kHttpResponseEmpty));
+    int32_t depth = -1;
+    auto index_response =
+        olp::dataservice::read::QueryApi::QuadTreeIndexVolatile(
+            *client_, kLayerid, kQuadKey, depth, boost::none, boost::none,
+            olp::client::CancellationContext{});
+
+    ASSERT_FALSE(index_response.IsSuccessful());
+    auto error = index_response.GetError();
+    ASSERT_EQ(http::HttpStatusCode::BAD_REQUEST, error.GetHttpStatusCode());
+    ASSERT_EQ(kHttpResponseEmpty, error.GetMessage());
+  }
+  {
+    SCOPED_TRACE("Empty additional fields.");
+    EXPECT_CALL(*network_mock_, Send(_, _, _, _, _))
+        .WillOnce(ReturnHttpResponse(http::NetworkResponse().WithStatus(
+                                         http::HttpStatusCode::BAD_REQUEST),
+                                     kHttpResponseEmpty));
+
+    int32_t depth = 2;
+    std::vector<std::string> empty_fields;
+    auto index_response =
+        olp::dataservice::read::QueryApi::QuadTreeIndexVolatile(
+            *client_, kLayerid, kQuadKey, depth, empty_fields, boost::none,
+            olp::client::CancellationContext{});
+
+    ASSERT_FALSE(index_response.IsSuccessful());
+    auto error = index_response.GetError();
+    ASSERT_EQ(http::HttpStatusCode::BAD_REQUEST, error.GetHttpStatusCode());
+    ASSERT_EQ(kHttpResponseEmpty, error.GetMessage());
+  }
+  {
+    SCOPED_TRACE("Empty billing tag.");
```

Comment:
Billing tag is optional, you will not get any error from OLP if this is empty. Or do you?

Reply:
Yes, its optional, so boost::none is ok, but empty string is invalid tag. Rest API says it should be at least 4 symbols.

Reply:
Ok, then we can keep the test also. Nice!